### ISSUE TITLE
Add text argument to lintr::lint

### ISSUE THREAD

**Issue by renkun-ken** [jimhester/lintr#475]
In languageserver, we have to pass the file content to `lintr::lint()` (since the file is often being edited but not saved, and we do provide live diagnostics on user input) but also want it to behave as if the content is from a filename so that the following cases could work:

* `linter_file` is respected (https://github.com/REditorSupport/languageserver/issues/235)
* `object_usage_linter` detects its package (https://github.com/REditorSupport/languageserver/issues/238)
* Exclusions in `.lintr` not working (https://github.com/REditorSupport/languageserver/issues/282)

One bad solution is to write a temp file inside the user workspace (e.g. original file `test.R` and we write a temp `test.Rtmp`) but Language Server Protocol seems to discourage any change in user workspace directly rather than via LSP communication.

Therefore, I'm wondering if we could add a `text` argument so that we could use 

```r
lintr::lint(filename, linters = NULL, cache = FALSE, ..., parse_settings = TRUE, text = NULL)
```

by supplying both `filename` and `text`:

```r
lintr::lint(filename, text = text)
```

so that both are respected and `lint` wouldn't have to read the file again if `text` is supplied.

**Comment by jimhester** [comment]
Sure, seems reasonable, you can open a PR with the necessary changes.

**Comment by jimhester** [comment]
Also @renkun-ken if you are interested in [contributing more](https://github.com/jimhester/lintr/issues/318#issuecomment-389972589) to lintr I can invite you as a collaborator.

**Comment by renkun-ken** [comment]
@jimhester  I could make a PR for this, and of course I'm interested in contributing more to lintr as a collaborator. Thanks!
